### PR TITLE
[IA-2764] Create GCE OS image based off a COS image with right Docker container images cached

### PR DESCRIPTION
**Pull request by kyuksel**
[JIRA ticket](https://broadworkbench.atlassian.net/browse/IA-2764)

This PR
- updates `Jenkinsfile` to produce GCE OS images instead of snapshots since the latter turned out to not work for us.
- fixes the Welder and AoU container image versions.

It was used to successfully build [custom-leo-gce-cos-image-704c032](https://console.cloud.google.com/compute/imagesDetail/projects/broad-dsp-gcr-public/global/images/custom-leo-gce-cos-image-704c032?project=broad-dsp-gcr-public).

### CODE REVIEW

File: jenkins/gce-custom-images/prepare_gce_image.sh [hunk 1]
```diff
@@ -17,17 +17,17 @@ set -e -x
...
-terra_jupyter_base="us.gcr.io/broad-dsp-gcr-public/terra-jupyter-base:0.0.19"
-terra_jupyter_python="us.gcr.io/broad-dsp-gcr-public/terra-jupyter-python:0.1.1"
-terra_jupyter_r="us.gcr.io/broad-dsp-gcr-public/terra-jupyter-r:1.0.13"
-terra_jupyter_bioconductor="us.gcr.io/broad-dsp-gcr-public/terra-jupyter-bioconductor:1.0.13"
-terra_jupyter_gatk="us.gcr.io/broad-dsp-gcr-public/terra-jupyter-gatk:1.1.1"
-terra_jupyter_aou_old="us.gcr.io/broad-dsp-gcr-public/terra-jupyter-aou:1.0.23"
-terra_jupyter_aou="us.gcr.io/broad-dsp-gcr-public/terra-jupyter-aou:1.1.2"
-
-welder_server="us.gcr.io/broad-dsp-gcr-public/welder-server:6cfad83"
+terra_jupyter_base="us.gcr.io/broad-dsp-gcr-public/terra-jupyter-base:0.0.20"
```
Comment (Qi77Qi) [inline jenkins/gce-custom-images/prepare_gce_image.sh:20]: thought we were saying we're removing these defaults?

Comment (rtitle) [reply jenkins/gce-custom-images/prepare_gce_image.sh:20]: We removed the Jenkins parameter defaults from `Jenkinsfile`; I think the versions in the `.sh` scripts get replaced regardless

Comment (kyuksel) [reply jenkins/gce-custom-images/prepare_gce_image.sh:20]: What Rob said…

File: http/src/main/resources/reference.conf [hunk 2]
```diff
@@ -85,8 +85,7 @@ dataproc {
 }
 
 gce {
-  customGceImage = "projects/broad-dsp-gcr-public/global/images/custom-leo-image-gce-debian9-5e2206c"
-  customGceSnapshot = "projects/broad-dsp-gcr-public/global/snapshots/custom-leo-gce-cos-snapshot-d1a330f"
+  customGceImage = "projects/broad-dsp-gcr-public/global/snapshots/custom-leo-gce-cos-image-704c032"
```
Comment (rtitle) [inline http/src/main/resources/reference.conf]: Is updating this image to COS dependent on @Qi77Qi's PR?

(Should we stick with the `debian9` image in this PR so we can merge it first? 🤔 )

**Comment by kyuksel** [comment]
unrelated leonardo-build failure

jenkins-retest

File: http/src/main/resources/reference.conf (hunk 2 above)
Comment (kyuksel) [reply http/src/main/resources/reference.conf]: I thought the part of Qi's work this image version depends on was merged but please let me know if that's not the case @Qi77Qi and I'll change it here.

File: jenkins/gce-custom-images/prepare_gce_image.sh (hunk 1 above)
Comment (kyuksel) [reply jenkins/gce-custom-images/prepare_gce_image.sh:20]: Speaking of `prepare….sh` scripts getting replaced…due to the issues mentioned in a [new ticket](https://broadworkbench.atlassian.net/browse/IA-2770?atlOrigin=eyJpIjoiOWVhNzMwNDIyMTNjNDA2YmIzYmQwZWU5MmJmMzA3NzQiLCJwIjoiamlyYS1zbGFjay1pbnQifQ) that seem to predate any COS/GPU related work, `welder` and `aou` image versions will not be updated in `prepare….sh` scripts so we need to manually update the `prepare….sh` files when those two image versions change.

File: http/src/main/resources/reference.conf (hunk 2 above)
Comment (Qi77Qi) [reply http/src/main/resources/reference.conf]: I think this won't work. Using COS based images require a bunch changes done in my PR

Comment (rtitle) [reply http/src/main/resources/reference.conf]: Yeah I think [this PR](https://github.com/DataBiosphere/leonardo/pull/2088) is needed in order for COS to work.

File: jenkins/gce-custom-images/prepare_gce_image.sh (hunk 1 above)
Comment (rtitle) [reply jenkins/gce-custom-images/prepare_gce_image.sh:20]: Good to know! I knew the aou ones didn't get replaced but not welder.

File: http/src/main/resources/reference.conf (hunk 2 above)
Comment (kyuksel) [reply http/src/main/resources/reference.conf]: Cool - the integration tests failed for that reason too. Will revert.

**Comment by kyuksel** [comment]
I also just finished running the Jenkins job again, which successfully created a custom Dataproc image with all of the latest Docker container versions, but it failed to PR it because of git branch collision with the previous run's branch. I'll add the hash update to this PR.

File: http/src/main/resources/reference.conf [hunk 3]
```diff
@@ -63,7 +63,7 @@ dataproc {
   }
   # This image is used for legacy jupyter image where hail is compatible with earlier version of spark
   legacyCustomDataprocImage = "projects/broad-dsp-gcr-public/global/images/custom-leo-image-dataproc-1-2-79-debian9-5e2206c"
-  customDataprocImage = "projects/broad-dsp-gcr-public/global/images/custom-leo-image-dataproc-1-4-51-debian10-5e2206c"
```
Comment (Qi77Qi) [inline http/src/main/resources/reference.conf:66]: this is a non cos image based one?

**Review by Qi77Qi — APPROVED** [review]

**Review by rtitle — APPROVED** [review]
LGTM if tests pass.. I double checked the images in `reference.conf` 👍

**Pull request merged**